ebmbot/scheduler.py

```diff
@@ -0,0 +1,211 @@
+import json
+from datetime import datetime, timedelta, timezone
+
+from .connection import get_connection
+from .logger import log_call
+
+
+@log_call
+def schedule_job(type_, args, slack_channel, delay_seconds=0):
+    """Schedule job to be run.
+
+    Only one job of any type may be scheduled.  If a job is already scheduled
+    but isn't running yet and another job of the same type is scheduled, the
+    record of the first job is updated.
+
+    If a job is already running, another job of that type may be scheduled.
+    """
+
+    conn = get_connection()
+
+    sql = """
+    SELECT id, started_at IS NOT NULL AS has_started
+    FROM job
+    WHERE type = ?
+    ORDER BY has_started
+    """
+
+    start_after = _now() + timedelta(seconds=delay_seconds)
+    args = json.dumps(args)
+
+    existing_jobs = list(conn.execute(sql, [type_]))
+
+    if len(existing_jobs) == 0:
```

Comment:
I suppose technically there's a race condition here but I don't think we care in practice.

I was thinking about how we'd avoid that if we did care about it, and I wondered about using separate tables for scheduled and running jobs, both with a unique constraint on `job_type` and then the dispatcher could do a delete and insert inside a transaction.

But anyway, all overkill as I don't think this will cause us any problems.